Source/WebCore/fileapi/BlobBuilder.cpp

```diff
@@ -71,7 +71,7 @@ void BlobBuilder::append(RefPtr<Blob>&& blob)
 
 void BlobBuilder::append(const String& text)
 {
-    auto bytes = PAL::UTF8Encoding().encode(text, PAL::UnencodableHandling::Entities, PAL::NFCNormalize::No);
+    auto bytes = PAL::TextCodecUTF8::encodeUTF8(text);
```

Comment:
How do we know for sure text is valid UTF8?

Reply:
I don't know exactly where we handle it, but these two code paths are exactly the same. In particular see how `TextCodecUTF8::encode` ends up calling `encodeUTF8` without these arguments.

Reply:
I would like to know where we handle lone surrogates in the string argument though. It has to be somewhere as otherwise all kinds of tests would start failing.